libs/ts/config-types/src/node-config/types.ts

```diff
@@ -39,15 +39,22 @@ const PublishingCriteriaSchema = S.Struct({
   peg_tolerance_percentage: S.Number,
 });
 
+const ContractSchema = S.Struct({
+  name: S.String,
+  address: S.NullishOr(S.String),
+  creation_byte_code: S.NullishOr(S.String),
+  deployed_byte_code: S.NullishOr(S.String),
+  min_quorum: S.NullishOr(S.Number),
+  contract_version: S.Number,
+});
+
 const ProviderSchema = S.Struct({
   private_key_path: S.String,
   url: S.String,
-  contract_address: S.NullishOr(S.String),
-  safe_address: S.NullishOr(S.String),
-  safe_min_quorum: S.Number,
   event_contract_address: S.NullishOr(S.String),
   multicall_contract_address: S.NullishOr(S.String),
-  transaction_retries_count_limit: S.Number,
+  transaction_retries_count_limit: S.NullishOr(S.Number),
+  transaction_retries_count_before_give_up: S.NullishOr(S.Number),
```

Comment:
These can not be nullish I think.

https://github.com/blocksense-network/blocksense/blob/714c668e6f3bbcc5af4d809ec1cf96e374c7d6c3/apps/sequencer/src/providers/provider.rs#L82-L83

Reply:
I agree, only one of them should be provided. I am not sure how to specify that in `types.ts`